custom_components/foxess_modbus/entities/modbus_charge_period_sensors.py

```diff
@@ -100,8 +110,7 @@ def native_value(self):
             and not _is_force_charge_enabled(value, other_value)
         ):
             value = self._last_enabled_value
-        hours, minutes = _parse_time(value)
-        value = time(hour=hours, minute=minutes)
+        value = parse_time_value(value)
```

Comment:
This seems a little back-to-front now? Wouldn't it make more sense to use `validation.py` instance of `Time`?

Reply:
To be honest, I don't particularly like using the validation stuff to check for valid times. The time-related sensors rely on the time being valid to avoid throwing exceptions, so making that validation "configurable" smells.

I tried to keep all of the logic for validating, parsing and serializing time registers inside modbus_charge_period_sensors.py -- validating them requires the same amount of knowledge as parsing/serializing them, and putting that validation logic inside validation.py just duplicates that knowledge around the place.

Reply:
Fair enough - if we were dealing with multiple instances of time from various sources it'd probably be a different story but I can see the logic in keeping everything together. :shipit:

Reply:
True, fair enough. In that case we'd probably want to move the time stuff into its own place?